src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -112,7 +112,7 @@ public DatabaseItemType GetNextType()
                     _log.Operations(msg);
                 _result.AddWarning(msg);
 
-                SkipArray(onSkipped: null, token: CancellationToken.None);
+                SkipArray(onSkipped: null, MaySkipBolb, CancellationToken.None);
```

Comment:
Apply all necessary comments from 5.0 here first. Then I will do 4.2 to 5.0 merge and you can do further adjustments.